server/router_utils.go

```diff
@@ -0,0 +1,126 @@
+package server
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"net/http"
+	"strconv"
+
+	"github.com/RedHatInsights/insights-operator-utils/responses"
+	"github.com/RedHatInsights/insights-results-aggregator/types"
+	"github.com/google/uuid"
+	"github.com/gorilla/mux"
+)
+
+// RouterMissingParamError missing parameter in URL
+type RouterMissingParamError struct {
+	paramName string
+}
+
+func (e *RouterMissingParamError) Error() string { return fmt.Sprintf("missing param %v", e.paramName) }
+
+// RouterParsingError parsing error, for example string when we expected integer
+type RouterParsingError struct {
+	paramName  string
+	paramValue interface{}
+	errString  string
+}
+
+func (e *RouterParsingError) Error() string {
+	return fmt.Sprintf(
+		"Error during parsing param %v with value %v. Error: %v",
+		e.paramName, e.paramValue, e.errString,
+	)
+}
+
+// GetRouterParam retrieves parameter from URL like `/organization/{org_id}`
+func GetRouterParam(request *http.Request, paramName string) (string, error) {
+	value, found := mux.Vars(request)[paramName]
+	if !found {
+		return "", &RouterMissingParamError{paramName: paramName}
+	}
+
+	return value, nil
+}
+
+// GetRouterIntParam retrieves parameter from URL like `/organization/{org_id}`
+// and check it for being valid integer, otherwise returns error
+func GetRouterIntParam(request *http.Request, paramName string) (int64, error) {
```

Comment:
Is this should be exported? Because it used only in package `server`

Reply:
@bond95 fixed